fix: "Stores" field to not show loading spinner (#4669)

WildOrangutan opened the pull request:
### What
"simple input text field " was showing loading spinner, even if there were no autocomplete suggestions available.

Any input field, without autocomplete enabled, will now not show loading spinner.

### Screenshot
[untitled.webm](https://github.com/openfoodfacts/smooth-app/assets/5972966/66c62fbf-bd4c-4511-9e75-ea7817ca9904)

### Fixes bug(s)
Fixes #4669

WildOrangutan committed:
fix: Loading when no suggestions (#4669)

Loading spinner was showing, even if there were no loading or
suggestions available.

Fixes #4669

monsieurtanuki commented on packages/smooth_app/lib/pages/product/simple_input_text_field.dart:
```diff
@@ -79,6 +79,7 @@ class _SimpleInputTextFieldState extends State<SimpleInputTextField> {
       if (_manager == null || search.length < widget.minLengthForSuggestions) {
         _suggestions[search] = _SearchResults.empty();
       } else {
+        _setLoading(true);
```
What about `_setLoading(false);` in the `catch`?

WildOrangutan replied:
I would say no, since there is a `_setLoading(false)` call, that will be always called after try-catch block.

But there is a bug here. If exception is thrown, `_suggestions[search]` will be null, and it will not disable loading and also crash the app on the last return. Will fix it.

WildOrangutan committed:
refactor: Simple input text field loading

Also fixes crash and properly disables loading, when try-catch block throws exception.

monsieurtanuki reviewed (approved):
Hi @WildOrangutan!
I haven't tested it but that looks good to me, thank you!
